Use image listing from registry-image-widgets

stefwalter opened the pull request:
This shares code with the Registry Console and Openshift Web Console for the ImageStream image listing.

stefwalter commented:
Openshift Web Console View:

![screenshot from 2017-01-18 20-55-26](https://cloud.githubusercontent.com/assets/795070/22080629/8d3dee24-ddc0-11e6-8ff6-c922fa594456.png)

stefwalter commented:
Existing Registry Console Views:

![screenshot from 2017-01-18 20-54-31](https://cloud.githubusercontent.com/assets/795070/22080652/a7754238-ddc0-11e6-8dd5-54753da853bc.png)

![screenshot from 2017-01-18 20-54-36](https://cloud.githubusercontent.com/assets/795070/22080627/8d1d0fec-ddc0-11e6-95ed-66f08120eaf8.png)

stefwalter commented:
CC @andreasn @jwforres

stefwalter commented:
CC @dperpeet

spadgett commented:
Very cool

jwforres commented:
I see this is pulling in gettext now @stefwalter is this going to cause the image stream bits to be translated when the rest of the console is not? or does something have to be enabled for gettext to use a different language other than the default

stefwalter commented:
> I see this is pulling in gettext now @stefwalter is this going to cause the image stream bits to be translated when the rest of the console is not? or does something have to be enabled for gettext to use a different language other than the default.

It won't. The because project does not initiallize angular-gettext ... nor does the project load any translations.

jwforres commented:
OK sounds good, I will pull this down today and try it out

On Jan 19, 2017 2:49 AM, "Stef Walter" <notifications@github.com> wrote:

> I see this is pulling in gettext now @stefwalter
> <https://github.com/stefwalter> is this going to cause the image stream
> bits to be translated when the rest of the console is not? or does
> something have to be enabled for gettext to use a different language other
> than the default.
>
> It won't. The because project does not initiallize angular-gettext ... nor
> does the project load any translations.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin-web-console/pull/1136#issuecomment-273704137>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABZk7WEMyYkGIvOTvcaF8sAhqScOeHIvks5rTxVpgaJpZM4LnUmW>
> .
>

jwforres commented:
When no tags have been pushed it says "No image streams are present"  seems like that should say "No tags are present"

![image-stream-1](https://cloud.githubusercontent.com/assets/1467629/22111495/8212efb4-de2d-11e6-92c6-4cd1d746e98e.png)


When no tags have been pushed the Image count is empty instead of saying '0'. Also noticed it is inconsistent on this screen whether the labels have a ':' after them.  The rest of the console does have ':' after any labels.

![image-stream-2](https://cloud.githubusercontent.com/assets/1467629/22111496/8218e234-de2d-11e6-8c16-2d4b353234c4.png)



This version of the image widgets has some kind of issue with the image layers widget, when i revert back to master and use the image widgets version that points to this same image shows up just fine in the layers widget:

![image-stream-3](https://cloud.githubusercontent.com/assets/1467629/22111497/821c5efa-de2d-11e6-9866-30378432f751.png)

jwforres commented:
When I hover over the dropdown arrow it messes up the row highlighting, part of the row is gray and part of the row is blue.

![image-stream-4](https://cloud.githubusercontent.com/assets/1467629/22111677/2b4d8620-de2e-11e6-89de-c512819af249.png)

jwforres commented:
The "pushed from" in the originates from column is not on the same horizontal as the tag and the SHA, which creates a strange white gap, seems like it should move up.

![image-stream-5](https://cloud.githubusercontent.com/assets/1467629/22111761/7a7a3932-de2e-11e6-90ee-653c71242cda.png)

jwforres commented:
The text "To push to an image to this image stream:"  isnt right, should be "To push an image to this image stream:"

jwforres commented:
What is expanding a tag supposed to do?  I thought maybe it was supposed to show the history for that tag, but when I expand it then it isn't showing me anything different, even when I know a tag has had multiple images pushed to it.  It just seems to add a line between the sha and the timestamp.

jwforres commented:
The sha causes problems at various responsive sizes, pushing content off screen. It needs to have word-break on it

![image-stream-6](https://cloud.githubusercontent.com/assets/1467629/22112124/d8be6756-de2f-11e6-9a62-cc80709d20a7.png)

jwforres commented:
I'm actually seeing similar issues on the image widget page with long identifiers pushing off the page at mobile

jwforres commented:
When no images are pushed yet, the table doesn't extend all the way to the right of the screen, as soon as there as an image it does

![image-stream-7](https://cloud.githubusercontent.com/assets/1467629/22112267/48f70e4c-de30-11e6-8f1f-1f3a12d356af.png)

andreasn commented:
There is a couple of things I'm not currently satisfied about how this view works, also in the registry.
 * Tags appearance looks out of place. Maybe this would do well from using the badge look. http://www.patternfly.org/pattern-library/widgets/#badges
 * The rows are very tall.
 * The list view looks and works a bit different from both Cockpit and Patternfly (mostly because it's older code from before we refined the appearance of things) http://www.patternfly.org/pattern-library/content-views/list-view/#/expanding-rows
 * The SHA string is reeeeally long and just breaks the rows apart. This will only get worse with SHA512. I understand the idea about not hiding part of it (opening the question of "will I be able to copy this out?"), but maybe we can ellipsis it a bit and then show the full value on hover.

I'll put together a suggestion in a mockup for how to solve these.

stefwalter commented:
Thanks for the review @andreasn. May I suggest changes as a follow up pull request, and the first one gets the two projects mostly in sync on the view? Are you okay with that @jwforres 

@jwforres and @andreasn: The inline expansion stuff wasn't supposed to show up in the Openshift Web Console. That's a bug.

@andreasn In the registry the entire sha256 is necessary as the docker command line doesn't support abbreviated SHA identifiers for images. But perhaps that's a place where the Openshift Web Console can diverge and the implementation can take a CSS option to truncate this ... just thinking outloud.

andreasn commented:
I'm OK with doing it as a followup if @jwforres is.

stefwalter commented:
> When no tags have been pushed it says "No image streams are present" seems like that should say "No tags are present"

Fixed.

> When no tags have been pushed the Image count is empty instead of saying '0'.

Fixed.

> Also noticed it is inconsistent on this screen whether the labels have a ':' after them. The rest of the console does have ':' after any labels.

The Registry Console doesn't use colons, and uses different styling for metadata/data. So to make it consistent I've just added these colons via CSS in the Openshift Web Console use of the widgets.

> This version of the image widgets has some kind of issue with the image layers widget, when i revert back to master and use the image widgets version that points to this same image shows up just fine in the layers widget:

Fixed. This was a regression from when we marked is translatable. Now works.

> When I hover over the dropdown arrow it messes up the row highlighting, part of the row is gray and part of the row is blue.

This doesn't happen for me. Maybe I tagged the bower registry-image-widgets build incorrectly or something. Should work now. The dropdown arrow is not part of the changes we intended to share with the Openshift Web Console. The Registry Console has a listing pattern that allows expansion of data inline, but that doesn't need to be carried over into the Openshift Web Console.

> The "pushed from" in the originates from column is not on the same horizontal as the tag and the SHA, which creates a strange white gap, seems like it should move up.

I'll leave this up to @andreasn's follow up pull request.

> The text "To push to an image to this image stream:" isnt right, should be "To push an image to this image stream:"

Fixed.

> When no images are pushed yet, the table doesn't extend all the way to the right of the screen, as soon as there as an image it does

Fixed.

> The sha causes problems at various responsive sizes, pushing content off screen. It needs to have word-break on it

Added word-break.

stefwalter commented:
Everything fixed except for the row redesign, which should be done as a follow up to getting this feature "in" before the freeze.

pweil- commented:
@stefwalter please rebase

@jwforres @spadgett bump for another review since a lot of things have been fixed :+1:

stefwalter commented:
Rebased.

stefwalter commented:
However origin-web-console no longer authenticates my version of openshift, and I'm running out of time to sort that all out.

```
[root@f1 ~]# openshift version
openshift v1.4.1+3f9807a
kubernetes v1.4.0+776c994
etcd 3.1.0-rc.0
```

jwforres commented:
it is probably because of the change to remove localhost:9000 from the
default set of redirectURIs allowed in the master for the console oauth
client.  As a cluster admin you need to run

oc patch oauthclient/openshift-web-console -p '{"redirectURIs":["
https://localhost:9000/"]}'

We all use oc cluster up for development, which runs that step for us.

On Mon, Feb 6, 2017 at 5:36 AM, Stef Walter <notifications@github.com>
wrote:

> However origin-web-console no longer authenticates my version of
> openshift, and I'm running out of time to sort that all out.
>
> [root@f1 ~]# openshift version
> openshift v1.4.1+3f9807a
> kubernetes v1.4.0+776c994
> etcd 3.1.0-rc.0
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin-web-console/pull/1136#issuecomment-277644586>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABZk7UlKiZs9Lor-Cmap9Ys_JbL_My19ks5rZvecgaJpZM4LnUmW>
> .
>

stefwalter commented:
Thanks @jwforres. That did the trick. Smoke testing done.

andreasn commented:
I've opened an issue in Cockpit for the changes I

petervo commented:
@jwforres opened https://github.com/openshift/registry-image-widgets/pull/7 with the redesign changes. Once that gets in i'll update this PR.

jwforres commented:
@petervo thanks! merged the other PR and released v0.0.8

petervo commented:
@jwforres, Thanks, rebased and updated this one.

jwforres commented:
thanks @petervo 

I'm seeing two bugs currently

1) When you hover over a row the visual style is a little weird where the blue hover color appears on three sides of the tag but not on the left.  This one is lower severity and I'd be ok fixing it in a follow-up as long as we get an issue open to track it.
![hover-state](https://user-images.githubusercontent.com/1467629/27056200-2627d9a6-4f95-11e7-8021-52b1a2cb11ea.png)

2) Bad overflow can occur at mobile res in a couple places.  The table can overflow, and the docker pull commands can overflow.
![overflow-mobile](https://user-images.githubusercontent.com/1467629/27056233-478ce398-4f95-11e7-9609-362e7fc2684e.png)

jwforres commented on app/views/browse/imagestream.html:
```diff
@@ -44,105 +44,14 @@ <h1 class="contains-actions">
-                  </tr>
-                </tbody>
-              </table>
+              <registry-imagestream-body imagestream="imageStream">
```
sorry I didn't notice it before, but the switch to use registry-imagestream-body changes the behavior of how annotations are presented to the user, everywhere else in the UI we do not show these to the user unless they click a link to reveal them.  This is because annotations are typically intended for machine use and not end user facing.

petervo commented:
@jwforres, for the docker commands  do you want ellipsis or line wrapping with small sizes. Line wraping looks weird but is better for copy/paste. Is that ok?

jwforres commented on app/views/browse/imagestream.html:
```diff
@@ -44,105 +44,14 @@ <h1 class="contains-actions">
-                  <dt>Docker pull spec:</dt>
-                  <dd>{{imageStream.status.dockerImageRepository}}</dd>
-                </dl>
-                <annotations annotations="imageStream.metadata.annotations"></annotations>
```
for reference, this is the annotations directive we use today that handles the show/hide behavior plus how we format annotations when visible

jwforres commented:
agree wrapping is probably better.  i know we do multi-line code style in
other places and managed to make it look alright, one example is in
_pod-template.html L94, i suspect the "command" class was probably adding
some special multi-line stuff

On Mon, Jun 12, 2017 at 5:57 PM, petervo <notifications@github.com> wrote:

> @jwforres <https://github.com/jwforres>, for the docker commands do you
> want ellipsis or line wrapping with small sizes. Line wraping looks weird
> but is better for copy/paste. Is that ok?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin-web-console/pull/1136#issuecomment-307944937>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABZk7U3uoWlKRAw6WjXoUaUubDCOmo_zks5sDbQygaJpZM4LnUmW>
> .
>

petervo commented:
@jwforres, opened https://github.com/openshift/registry-image-widgets/pull/8 and built this with that so you can check this out first before merging the other one.

Once that is in i can do a rebuild

jwforres commented:
looks like the table still has an issue at iphone 5 res which is 320px width

the quick fix is in imagestream.html to wrap `<registry-imagestream-body>` in a `<div class="table-responsive" style="padding: 0 5px;">`

this will prevent the layout like the nav from getting messed up when the table overflows the page, it wraps the table in a scrollable content area with a border when at mobile resolutions

if you are fine just making that change in this PR then we can merge the PR to image-registry-widgets and cut that release, let me know

petervo commented:
@jwforres pushed an update that fixes it for me.

jwforres commented:
@petervo I have your latest changes but I can still reproduce it with a tag name that is long

![long-tag-name](https://user-images.githubusercontent.com/1467629/27087773-e9053962-5023-11e7-9691-9f4bef863dc8.png)

petervo commented:
@jwforres updated again.

jwforres commented:
thanks LGTM now, think we can merge the other PR

jwforres commented:
I'll LGTM the other PR and you can merge/release it if you are ready

petervo commented:
@jwforres, i don't have access to marge or deploy to npm. I'll see if can I track down someone who does.

jwforres commented:
@petervo ah k i can merge and create the tag for bower, i dont think i have the npm access but bower is all we need to the console

jwforres commented:
@petervo ok 0.0.9 should be available for bower now

jwforres commented:
[merge]
